src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -206,21 +211,35 @@ public bool IsBackground
             }
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private extern bool GetIsBackground();
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "ThreadNative_GetIsBackground")]
+        private static partial Interop.BOOL GetIsBackground(ThreadHandle t);
```

Comment:
I noticed different QCalls using `Interop.BOOL` and `[return: MarshalAs(UnmanagedTypes.Bool)]`. Do we have a preference for this? I have a branch to unify them to `MarshalAs`.

Reply:
It isn't so simple as unifying them. There will be performance impact. Blindly converting from `Interop.BOOL` to `bool` will break the naturally blittable P/Invoke and can change a lot. In this case, I was simply being consistent. My preference would be to change to `bool` only if one of the other arguments is non-blittable _or_ there isn't any performance concern.

Reply:
I've never expected this to have performance impact for simple cases. When the parameters are simple values, the source generated marshalling code and `!= 0` should be inlined, and gets the same codegen with `!= Interop.BOOL.False`.

Reply:
> I've never expected this to have performance impact for simple cases.

I'm not sure what this means in this case. I don't see how one can make that sort of statement about inlining when there are so many variables that could impact it - number of arguments, size of arguments, etc. Instead I would apply the above logic for now and we can iterate on your PR if there are places where you think that should change.